src/Version.h

```diff
@@ -15,8 +17,14 @@
 
 namespace Version {
 
+using exact_width_mumble_raw_version_t = std::uint32_t;
+using mumble_raw_version_t             = unsigned int;
+
+static_assert(sizeof(exact_width_mumble_raw_version_t) <= sizeof(mumble_raw_version_t),
+			  "Invalid scalar type used to represent Mumble versions");
```

Comment:
Because forcing integer widths that don't conform to the machine's native integer size is inefficient and in most cases not needed. So here we only ensure that the native integer size is big enough to hold at least a uint32 but may be larger

Reply:
`unsigned int` is not guaranteed to be the "native integer size". Besides, it's good practice to use fixed-width types, as commonly seen in modern APIs.

Requesting a greater width than needed can cause greater memory consumption, especially if the variable is in a structure; padding is added to achieve proper alignment, unless you pack it (required in very specific cases, e.g. protocol messages).

I strongly recommend to check the produced assembly before making a decision when it comes to that kind of optimization, which you can do quickly using online tools such as https://godbolt.org.

Reply:
> unsigned int is not guaranteed to be the "native integer size".

Maybe not guaranteed but very likely. Definitely more likely than when specifying a precise integer width

> Besides, it's good practice to use fixed-width types, as commonly seen in modern APIs.

It's not an API though and outside of APIs, I consider it a very bad practice, since it clutters the source code with unnecessary information (the width of the used integer type).

> Requesting a greater width than needed can cause greater memory consumption, especially if the variable is in a structure; padding is added to achieve proper alignment, unless you pack it (required in very specific cases, e.g. protocol messages).

Which in our use-case would probably result in a few extra byte of total memory consumption, which nobody will ever notice. Besides: Oftentimes these additional memory consumption can lead to improved runtime speed, because the computer doesn't have to transform the integer width.

> I strongly recommend to check the produced assembly

Nah, I'll not start digging up any assembly. I don't know Assembly and I don't intend to learn it. Especially not for something as trivial as this.